### ISSUE TITLE
Is train_rm not supported by multiple GPUs?

### ISSUE THREAD

**Issue by OceannTwT** [hiyouga/ChatGLM-Efficient-Tuning#44]
RT

**Comment by OceannTwT** [comment]
> We suppose that train_rm and train_ppo support distributed training using the Accelerate library, but it is yet to be validated.

I tested just before and utilized deepspeed to make it support on multiple GPUs, I think you are right that it is also supported by Accelerate library(Just in my guess). Anyway, thanks for your reply~

**Comment by wesleyhuang2014** [comment]
I use the following command to train the reward model with multiple GPUs, it failed with runtime error. It works while only use one GPU, and  in ptuning v2, the command `CUDA_VISIBLE_DEVICES=0,1  accelerate launch --multi_gpu  src/finetune.py` works well. Could you please help to check it? Thanks.
``` Bash
CUDA_VISIBLE_DEVICES=0,1 accelerate launch --multi_gpu src/train_rm.py \
    --do_train \
    --dataset comparison_gpt4_en \
    --finetuning_type lora \
    --output_dir path_to_rm_checkpoint \
    --per_device_train_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --lr_scheduler_type cosine \
    --logging_steps 10 \
    --save_steps 1000 \
    --learning_rate 5e-5 \
    --num_train_epochs 1.0 \
    --fp16 \
    --ddp_find_unused_parameters False
```

**Comment by hiyouga** [comment]
@wesleyhuang2014 Sure, there were several bugs in distributed training of the reward model. I have fixed them in the latest version, please update with `git pull` and train with the above arguments. Note that reward model training is only compatible with LoRA tuning.
